chore: deprecate -h, only -f needed for list-devices

xavierchanth opened the pull request:
<!--
Please make sure you've read and understood our contributing guidelines in CONTRIBUTING.md

If this is a bug fix, make sure your description includes "fixes #xxxx", or
"closes #xxxx"

Please provide the following information:
-->

**- What I did**

- deprecated -h, --host (also hide both of these)
- made it so -t, and -h are not needed to --list-devices
- with -t only list-devices for that daemon, without -t, list for all devices.

**- How I did it**

**- How to verify it**

- Tested -h and --host for breaking changes, still negotiates with rv correctly.
- List devices works with -t, only listing devices for -t
- List devices work without -t, listing devices as devicename@deviceatsign
  - This approach maintains compatibility with the notes from #968 

**- Description for the changelog**
chore: deprecate -h, only -f needed for list-devices
<!--
Write a short (one line) summary that describes the changes in this
pull request for inclusion in the changelog:
-->

xavierchanth committed:
chore: deprecate -h, only -f needed for list-devices

xavierchanth commented on packages/dart/noports_core/lib/src/sshnp/models/sshnp_device_list.dart:
```diff
@@ -12,4 +12,9 @@ class SshnpDeviceList {
+  void add(SshnpDeviceList o) {
+    info.addEntries(o.info.entries);
+    activeDevices.addAll(o.activeDevices);
+  }
```
Helper function to join the entries of another SshnpDeviceList into this one

xavierchanth commented on packages/dart/noports_core/lib/src/sshnp/models/sshnp_params.dart:
```diff
@@ -262,16 +263,23 @@ class SshnpParams extends ClientParamsBase
+          (throw ArgumentError(
+              'Option to is mandatory, unless list-devices is passed.'));
+      partial.srvdAtSign ?? (throw ArgumentError('srvdAtSign is mandatory'));
+    }
```
Only throw argument errors if list-devices is also false.

xavierchanth committed:
fix: error message

xavierchanth commented on packages/dart/noports_core/lib/src/sshnp/models/sshnp_params.dart:
```diff
@@ -262,16 +263,25 @@ class SshnpParams extends ClientParamsBase
-      sshnpdAtSign: partial.sshnpdAtSign!,
-      srvdAtSign: partial.srvdAtSign!,
+      sshnpdAtSign: partial.sshnpdAtSign ?? "",
+      srvdAtSign: partial.srvdAtSign ?? "",
```
Easier to pass these as empty strings than to refactor everything to null check.

xavierchanth commented on packages/dart/noports_core/lib/src/sshnp/models/sshnp_params.dart:
```diff
@@ -479,7 +489,8 @@ class SshnpPartialParams {
           : AtUtils.fixAtSign(args[SshnpArg.toArg.name]),
-      srvdAtSign: args[SshnpArg.srvdArg.name],
+      srvdAtSign:
+          args[SshnpArg.srvdArg.name] ?? args[SshnpArg.legacySrvdArg.name],
```
map -h and --host as a fallback to -r and --srvd

xavierchanth commented on packages/dart/noports_core/lib/src/sshnp/util/sshnpd_channel/sshnpd_channel.dart:
```diff
@@ -255,39 +255,86 @@ abstract class SshnpdChannel with AsyncInitialization, AtClientBindings {
+    return consolidatedList;
+  }
 
+  Future<SshnpDeviceList> _listDevices(String sharedBy,
```
This is the old logic for listDevices

xavierchanth commented on packages/dart/noports_core/lib/src/sshnp/util/sshnpd_channel/sshnpd_channel.dart:
```diff
@@ -255,39 +255,86 @@ abstract class SshnpdChannel with AsyncInitialization, AtClientBindings {
   /// and corresponding info, and a list of active devices (devices which also
   /// responded to our real-time ping).
   Future<SshnpDeviceList> listDevices() async {
```
This function wraps the previous logic, deferring to the old logic if -t is set, or aggregating across all devices if -t is not set.

xavierchanth committed:
chore: restore a previous behaviour

xavierchanth committed:
chore: restore a previous behaviour

xavierchanth committed:
fix: typo

xavierchanth commented on packages/dart/noports_core/lib/src/sshnp/util/sshnpd_channel/sshnpd_channel.dart:
```diff
@@ -325,7 +372,8 @@ abstract class SshnpdChannel with AsyncInitialization, AtClientBindings {
       ..sharedWith = sharedWith
-      ..sharedBy = sharedBy
+      ..sharedBy =
+          sharedBy // for some reason, if this is null it only returns cached keys
```
Noted a behavioral bug, which explains why I went an round-about way to implement listDevices without -t

xavierchanth committed:
Merge branch 'trunk' into less-flags

cpswan committed:
Merge branch 'trunk' into less-flags

cpswan reviewed (commented):
LGTM, though can there be a new test for this path through the usage?

xavierchanth commented:
> LGTM, though can there be a new test for this path through the usage?

Sure, I will add it today

xavierchanth committed:
Merge branch 'trunk' into less-flags

xavierchanth committed:
chore: added e2e test for minus_r_flag

xavierchanth commented:
#### Additional changes
1. `minus_r_flag` e2e_all test:
- base `sshnp` command still uses `-h` otherwise we have to do a bunch of refactoring of the tests to change the base command based on the client version
- Test 1 : `--host` regression test
- Test 2 : `-r` overrides `-h` value (made `-h` value a non-activated atSign, so it will fail if the `-h` value is used)

2. Updated noports core to 6.0.7 and updated changelog
3. Updated sshnoports to 5.2.0 (since I need a cut-off version for when this test is allowed to run)

xavierchanth commented on tests/e2e_all/scripts/tests/minus_r_flag:
```diff
@@ -0,0 +1,125 @@
+# 2. Run with -h && -r (-h set to an invalid rvd atSign)
+#  - expect this test to pass
+
+# TODO: Would like to add a test for, but this causes the test to timeout waiting for the bad srvd
+# 3. Run with -h && -r (-r set to an invalid rvd atSign)
+#  - expect this test to fail
```
It would be nice to implement this test, but test case 2 asserts the inverse case is validated.

xavierchanth committed:
Merge branch 'trunk' into less-flags
